lib/Ushur.js

```diff
@@ -33,9 +32,12 @@ function Ushur(credentials) {
    * @type {String}
    */
   this.tokenId = undefined;
-
+  this.hostname = credentials.hostname || null;
   //Authenticate in the start
-  this.auth();
+  if(credentials.mode && credentials.mode.toLowerCase() == "promise")
```

Comment:
Can you also update readme regarding usage of hostname field with examples. And also usage of mode(promise)